.github/workflows/NightlyTests.yml

```diff
@@ -64,6 +65,325 @@ jobs:
        run: |
          python3 test/memoryleak/test_memory_leaks.py
 
+  regression-lto-benchmark-runner:
+     name: Benchmark runner lto vs non-lto (OSX)
```

Comment:
Can we remove all of these benchmark runs, or move them to a workflow that is only run when triggered (e.g. `ExtraTests.yml`)? While I think they are interesting to run once it seems like unnecessary bloating of the nightly tests otherwise.

Reply:
Thanks, I struggled on where to put them, since with some code there is an easier to follow blueprint on how to use them, it seemed useful, but extra file is for the best.

I moved those 4 tests to `ExtendedTests.yml` (mostly since I forgot you had made a precise suggestion). I can also just move the file to `ExtraTests.yml`, either works.